extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/JwtUtils.java

```diff
@@ -36,31 +36,28 @@
 import java.util.UUID;
 
 /**
- * Convenience/helper class to generate, verify and deserialize verifiable credentials, which are, in fact, Signed JSON Web Tokens (JWTs).
+ * Convenience/helper class to generate and verify Signed JSON Web Tokens (JWTs) for communicating between connector instances.
  */
-public class VerifiableCredentialFactory {
-
+class JwtUtils {
     // RFC 7519 Registered (standard) claims
     private static final String ISSUER_CLAIM = "iss";
+    private static final String SUBJECT_CLAIM = "sub";
     private static final String EXPIRATION_TIME_CLAIM = "exp";
 
-    // Subject claim value
-    public static final String VERIFIABLE_CREDENTIAL = "verifiable-credential";
-
     /**
-     * Creates a signed JWT {@link SignedJWT} that contains a set of claims and an issuer. Although all private key types are possible, in the context of Distributed Identity and ION
+     * Creates a signed JWT {@link SignedJWT} that contains a set of claims and an issuer. Although all private key types are possible, in the context of Distributed Identity
      * using an Elliptic Curve key ({@code P-256}) is advisable.
      *
-     * @param privateKey A Private Key represented as {@link PrivateKeyWrapper}.
-     * @param issuer     the "owner" of the VC, in most cases this will be the DID ID. The VC will store this in the "iss" claim
-     * @param audience   the audience of the token, e.g. the IDS Webhook address. The VC will store this in the "aud" claim
-     * @param clock      clock used to get current time
+     * @param privateKey       A Private Key represented as {@link PrivateKeyWrapper}.
+     * @param issuerAndSubject the value of the token issuer and subject claims. For Distributed Identity, these values are identical.
+     * @param audience         the value of the token audience claim, e.g. the IDS Webhook address.
+     * @param clock            clock used to get current time
      * @return a {@code SignedJWT} that is signed with the private key and contains all claims listed
      */
-    public static SignedJWT create(PrivateKeyWrapper privateKey, String issuer, String audience, Clock clock) {
-        var claimsSet = new JWTClaimsSet.Builder().issuer(issuer)
-                .issuer(issuer)
-                .subject(VERIFIABLE_CREDENTIAL)
+    static SignedJWT create(PrivateKeyWrapper privateKey, String issuerAndSubject, String audience, Clock clock) {
```

Comment:
nit: issuerAndSubject as per javadoc seems like here we always same value for issue and subject for now. But it looks little strange if we use one input argument for two different purposes. 

may we have different args here one for issue and one for subject and can have an overloaded method which have `String issuerAndSubject` for this use case.

But this more of my coding style choice so if you do not prefer it then we leave it. Thanks.

Reply:
Thanks, implemented along those lines